### PR TITLE
Draft Sim Implementation/Refactor

### PR DESCRIPTION
**Pull request by andrew032011**
### Summary
Sim Out Hardware Calls using a method `step()`, which runs FSW through one-time step. A dictionary is taken in as sensory input, which replaces all sensory hardware calls. A sim_output object is written to, and that replaces any hardware calls for actuation.

### Testing
TBD. We need to integrate the sim with flight software in order to fully test this.

### Notes
This pull request sets up the structure for integration from FSW, but in order to test this fully, we need to make some changes on the sim side as well. For now, I leave TODO comments for parts that are not fully defined on the Sim's side yet.

### Comments
- [x] Add an x between the brackets if you commented your code well!

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -83,6 +84,13 @@ def __init__(self):
         logging.info("opening UDP client socket")
         self.client = Client("192.168.0.200", 3333)
 
+        self.is_sim_run = is_sim_run
+
+        if self.is_sim_run:
+            logging.info("initializing sim data object")
+            self.sim_output = SimData()
+            self.sim_input = None
```
Comment (cameron-goddard) [inline main.py:92]: I'm assuming this won't be None in the future once the sim can send data to FSW?

**Review by cameron-goddard — APPROVED** [review]
Looking good so far

File: main.py (hunk 1 above)
Comment (andrew032011) [reply main.py:92]: Oh good point. That's redundant actually since I initialize it in the `step()` function. I'll remove the line.

**Review by adamrnasir — APPROVED** [review]
This looks great— thanks for the work!

**Pull request merged**